internal/pkg/aws/s3/s3.go

```diff
@@ -161,3 +175,24 @@ func ParseURL(url string) (bucket string, key string, err error) {
 	bucket, key = strings.Split(parsedURL[0], ".")[0], parsedURL[1]
 	return
 }
+
+// Check whether the bucket exists before proceeding with empty the bucket
+func (s *S3) isBucketExists(bucket string) (bool, error) {
+	input := &s3.HeadBucketInput{
+		Bucket: aws.String(bucket),
+	}
+	_, err := s.s3Client.HeadBucket(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+				case notFound:
+					return false, nil
+				default:
+					return false, aerr
+			}
+		}
+		return false, err
```

Comment:
nit: I believe these two statements return the same error, maybe we can simplify it like this:
```go
if aerr, ok := err.(awserr.Error); ok && aerr.Code() == notFound {
     return false, nil
}
return false, err
```

Reply:
Done